website/docs/language/style.mdx

```diff
@@ -0,0 +1,729 @@
+
+<CodeBlockConfig hideClipboard>
+
+```hcl
+variable "web_instances" {
+  type        = list(string)
+  description = "A list of instances for the web application"
+  default = [
+    "ui",
+    "api",
+    "db",
+    "metrics"
+  ]
+}
+
+resource "aws_instance" "web" {
+  for_each = toset(var.web_instances)
+
+  ami           = data.aws_ami.webapp.id
+  instance_type = "t3.micro"
+
+  tags = {
+    Name = "web_${each.key}"
+  }
+}
+
+output "web_private_ips" {
+  description = "Private IPs of the web instances"
+  value = {
+    for k, v in aws_instance.web : k => v.private_ip
+  }
+}
+
+output "web_ui_public_ip" {
+  description = "Public IP of the web UI instance"
+  value       = aws_instance.web["ui"].public_ip
+}
+```
+
+</CodeBlockConfig>
+
+The above example will create the following output:
+
+<CodeBlockConfig hideClipboard>
+
+```
+web_private_ips = {
+  "api" = "172.31.25.29"
+  "db" = "172.31.18.33"
+  "metrics" = "172.31.26.112"
+  "ui" = "172.31.20.142"
+}
+
+web_ui_public_ip = "18.216.208.182"
+```
+
+</CodeBlockConfig>
+
+The `count` meta-argument lets you create multiple instances of a resource from a single resource block. In the following example, if the value of `var.web_instance_count` is 2, the code will create two EC2 instances, one named `web_0` and the other named `web_1`.
+
+<CodeBlockConfig hideClipboard>
+
+```hcl
+resource "aws_instance" "web" {
+  count = var.web_instance_count
+
+  ami           = data.aws_ami.ubuntu.id
+  instance_type = "t3.micro"
+
+  tags = {
+    Name = "web_${count.index}"
+  }
+}
+```
+
+</CodeBlockConfig>
+
+A common practice to conditionally create resources is to use the `count` meta-argument with a [conditional expression](/terraform/language/expressions/conditionals). In the following example, Terraform will only create the `aws_instance` if `var.enable_metrics` is `true`.
+
+<CodeBlockConfig highlight="8" hideClipboard>
+
+```hcl
+variable "enable_metrics" {
+  description = "True if the metrics server should be deployed"
+  type        = bool
+  default     = true
+}
+
+resource "aws_instance" "web" {
+  count = var.enable_metrics ? 1 : 0
+
+  ami           = data.aws_ami.webapp.id
+  instance_type = "t3.micro"
+  ##...
+}
+```
+
+</CodeBlockConfig>
+
+We recommend that you use meta-arguments in moderation since you sacrifice explicit readability for an implicit understanding of how the configuration works. If the effect of the meta-argument is not immediately obvious, add a comment to make it clear to the reader.
```

Comment:
```suggestion

Meta-arguments simplify your code but add complexity, so use them in moderation since. If the effect of the meta-argument is not immediately obvious, use a comment for clarification.
```

Reply:
@ritsok I'm just a random passerby, but you left the word "since" at the end of the first sentence. 

Also, if you were to ask my opinion, this PR says "use in moderation" in this paragraph and above on line 29 as well, that's a really vague statement, and if you've worked on any moderately scaled Terraform or complex environment you will most likely regularly use for_each. Why should HCP suggest it's users to not to use for_each? Just my opinion though